### PR TITLE
fix: disable button focus style in Space.Compact

### PR DESCRIPTION
**Pull request by foryuki**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
Now in the v5 version, when a button is in the focus state, and then hover its adjacent left button, there is a problem with the right border of the left button. 

image

Supposed to be like:

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix button hover style in Space.Compact.      |
| 🇨🇳 Chinese |     修复 Space.Compact 下 Button hover 样式问题。      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
v5 里这个样式是不是应该用 outline？有设计规范么 @zombieJ

**Comment by zombieJ** [comment]
> v5 里这个样式是不是应该用 outline？有设计规范么 @zombieJ

这个贴边边线，outline 是 focus visible 投影，两个不太一样。投影部分因为 Safari 有 BUG，现在都是 box-shadow 模拟的。

File: components/select/style/index.tsx [hunk 1]
```diff
@@ -263,12 +263,10 @@ const genSelectStyle: GenerateStyle<SelectToken> = (token) => {
...
         },
         // Space.Compact
-        ...genCompactItemStyle(
-          token,
-          componentCls,
-          `${componentCls}-selector`,
-          `${componentCls}-focused`,
-        ),
+        ...genCompactItemStyle(token, componentCls, {
+          borderedElementCls: `${componentCls}-selector`,
```
Comment (zombieJ) [inline components/select/style/index.tsx:267]: 能省一点不，这个 key 不能被压缩

Comment (foryuki) [reply components/select/style/index.tsx:267]: 是说拼写太长了嘛？

Comment (zombieJ) [reply components/select/style/index.tsx:267]: 嗯呢~~

File: components/button/style/index.tsx [hunk 2]
```diff
@@ -69,7 +69,7 @@ const genSharedButtonStyle: GenerateStyle<ButtonToken, CSSObject> = (token): CSS
             display: 'inline-block',
             width: token.lineWidth,
             height: `calc(100% + ${token.lineWidth * 2}px)`,
-            backgroundColor: token.colorPrimaryBorder,
+            backgroundColor: token.colorPrimaryHover,
```
Comment (foryuki) [inline components/button/style/index.tsx:72]: @zombieJ 这里用伪元素实现的 primary button 之间的边框颜色可以改吗？原来的颜色看起来有点奇怪
Screenshot 2022-12-06 at 10 45 59 PM
改了之后效果：
Screenshot 2022-12-06 at 10 42 33 PM

Comment (zombieJ) [reply components/button/style/index.tsx:72]: 好看！

**Review by zombieJ — APPROVED** [review]

**Pull request merged**